### PR TITLE
refactor harmonyExportExpressionDependency to es6

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**

current tests should pass

**If relevant, link to documentation update:**

N/A

**Summary**

upgrade HarmonyExportExpressionDependency and HarmonyExportExpressionDependencyTemplate to es6

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: lib/dependencies/HarmonyExportExpressionDependency.js [hunk 1]
```diff
@@ -2,46 +2,55 @@
...
+			return;
+		}
+
 		source.replace(dep.rangeStatement[0], dep.rangeStatement[1] - 1, content);
 	}
-};
+
+	getContent(used) {
+		if(used) {
+			return `/* harmony default export */ exports[${JSON.stringify(used)}] = `;
```
Comment (Kovensky) [inline lib/dependencies/HarmonyExportExpressionDependency.js]: This is incorrect for `export default function` and `export default class` (either anonymous or named, but it's "less wrong" in the anonymous case). A getter needs to be used, similar to named exports, as they are mutable and, if anonymous, they need to receive the contextual name `default`.

It's a bug that was already there, though.

Comment (sokra) [inline lib/dependencies/HarmonyExportExpressionDependency.js]: `HarmonyExportExpressionDependency` is only used for expressions. `function a() {}` and `class A {}` are declaration so `HarmonyExportSpecifierDependency` is used.

Do you mean this?

``` js
export default function() {}
```

``` js
import f from "./file";
f.name === "default"
```

**Comment by timse** [comment]
anything left to do here for me :)?

**Comment by TheLarkInn** [comment]
Nope this is good thank you! @sokra and @Kovensky can continue discussion in new issue or same thread.

**Pull request merged**